proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,864 @@
+    space: space
+  })
+  ```
+
+* If space equals `lab` or `oklab`, let `changedColor` be the result of:
+
+  ```js
+  SassColor({
+    lightness: changedValue('lightness'),
+    a: changedValue('a'),
+    b: changedValue('b'),
+    alpha: changedValue('alpha'),
+    space: space
+  })
+  ```
+
+* If space equals `lch` or `oklch`, let `changedColor` be the result of:
+
+  ```js
+  SassColor({
+    lightness: changedValue('lightness'),
+    c: changedValue('c'),
+    h: changedValue('h'),
+    alpha: changedValue('alpha'),
+    space: space
+  })
+  ```
+
+* If `space` equals `a98-rgb`, `display-p3`, `prophoto-rgb`, `srgb`, or
+  `srgb-linear`, let `changedColor` be the result of:
+
+  ```js
+  SassColor({
+    red: changedValue('red'),
+    green: changedValue('green'),
+    blue: changedValue('blue'),
+    alpha: changedValue('alpha'),
+    space: space
+  })
+  ```
+
+* If `space` equals `xyz`,  `xyz-d50`, or `xyz-d65`, let `changedColor` be the
+    result of:
+
+  ```js
+  SassColor({
+    y: changedValue('y'),
+    x: changedValue('x'),
+    z: changedValue('z'),
+    alpha: changedValue('alpha'),
+    space: space
+  })
+  ```
+
+* If `initialSpace` is not equal to `space`, return the result of
+  [`changedColor.toSpace(initialSpace)`].
+
+* Otherwise, return `changedColor`.
+  
+[`this.space()`]: #space
+[`this.toSpace(space)`]: #tospace
+[`changedColor.toSpace(initialSpace)`]: #tospace
+[`Changing a Component Value`]: #changing-a-component-value
+
+```ts
+change(
+  options: {
+    [key in ChannelName]?: number | null;
+  } & {alpha?: number}
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameHSL]?: number | null;
+  } & {
+    alpha?: number;
+    space: ColorSpaceHSL;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameHWB]?: number | null;
+  } & {
+    alpha?: number;
+    space: ColorSpaceHWB;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameLAB]?: number | null;
+  } & {
+    alpha?: number | null;
+    space: ColorSpaceLAB;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameLCH]?: number | null;
+  } & {
+    alpha?: number | null;
+    space: ColorSpaceLCH;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameRGB]?: number | null;
+  } & {
+    alpha?: number | null;
+    space: ColorSpaceRGB;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameXYZ]?: number | null;
+  } & {
+    alpha?: number | null;
+    space: ColorSpaceXYZ;
+  }
+): SassColor;
+```
+
+### New Constructors
+
+* If `options.space` is not set, or `space` is a [legacy color space], follow
+  the previous procedure for [construction].
+
+* Otherwise, use the constructor that matches the value of `options.space`.
+
+[construction]: ../spec/js-api/value/color.d.ts.md#constructor
+
+#### LAB Channel Constructor
+
+Create a new SassColor in a color space with LAB channels -- `lab` and `oklab`.
+
+* Let `lightness` be the result of [parsing a channel value] with value
```

Comment:
Looks like this link reference isn't defined.

Reply:
Fixed in 3599c5bdd932eef6b28a1e9a2a969656694fc510